src/loaders/eventLoader.js

```diff
@@ -188,23 +163,73 @@ export const getAnalyticsRequest = async (
     return analyticsRequest;
 };
 
+export const getCount = async request => {
+    const d2 = await getD2();
+    return await d2.analytics.events.getCount(request);
+};
+
+export const loadData = async (request, config) => {
+    const d2 = await getD2();
+    const response = await d2.analytics.events.getQuery(request);
+
+    const names = {
+        ...(config.outputIdScheme !== 'ID' // TODO: Pass this through the the request to support ID/NAME/CODE output natively.  Server bugfix needed.
+            ? getApiResponseNames(response)
+            : null),
+        ...config.columnNames,
+    };
+
+    const data = response.rows
+        .map(row =>
+            createEventFeature(
+                response.headers,
+                names,
+                row,
+                config.eventCoordinateField
+            )
+        )
+        .filter(feature => isValidCoordinate(feature.geometry.coordinates));
+
+    return {
+        data,
+        names,
+        response,
+    };
+};
+
 // Include column for data element used for styling
 export const addStyleDataItem = (dataItems, styleDataItem) =>
     styleDataItem
         ? [
               ...dataItems,
-              styleDataItem && {
+              {
                   dimension: styleDataItem.id,
                   name: styleDataItem.name,
               },
           ]
         : [...dataItems];
 
-const createEventFeature = (headers, names, event, eventCoordinateField) => {
+const findIndex = (array, predicate) => {
+    var index = -1;
+    for (var i = 0; i < array.length; ++i) {
+        if (predicate(array[i], i, array)) {
+            index = i;
+            break;
+        }
+    }
+    return index;
+};
+
+export const createEventFeature = (
+    headers,
+    names,
+    event,
+    eventCoordinateField
+) => {
     const properties = event.reduce(
         (props, value, i) => ({
             ...props,
-            [headers[i].name]: names[value] || value,
+            [names[headers[i].name] || headers[i].name]: value,
```

Comment:
@turban I believe the code was incorrect here before since names comes from the header list. I updated it to work for Event Download but would love your review to make sure I didn't change the behavior of this function.